jdaviz/configs/default/plugins/export_plot/export_plot.py

```diff
@@ -89,26 +90,37 @@ def save_figure(self, filename=None, filetype=None):
             Filetype (PNG or SVG).  If `None`, will default based on filename or to PNG.
 
         """
+        if filename is not None:
+            filename = Path(filename)
+
         if filetype is None:
-            if filename is not None and '.' in filename:
-                filetype = filename.split('.')[-1]
+            if filename is not None and filename.suffix:
+                filetype = filename.suffix[1:].lower()
             else:
-                # default to png
-                filetype = 'png'
+                filetype = "png"  # default to png
 
         viewer = self.viewer.selected_obj
+
         if filetype == "png":
             if filename is None:
                 viewer.figure.save_png()
             else:
+                if not filename.parent.exists():
+                    raise ValueError(f"Invalid path={filename.parent}")
+
                 # support writing without save dialog
                 # https://github.com/bqplot/bqplot/pull/1397
                 def on_img_received(data):
-                    with open(os.path.expanduser(filename), 'bw') as f:

```

Comment:
@kecnry , I don't remember why you added `expanduser` here. What was the use case that triggered this? Using Path, I am hoping we don't need it anymore but should make sure.

Reply:
Nvm... Found the original issue https://github.com/spacetelescope/jdaviz/pull/929#issuecomment-1550331365